### PR TITLE
ENH strict channel priority for linux

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR turns on strict channel priority for linux on all builds. 

cc @isuruf @ocefpaf @conda-forge/core

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
Great @xhochy! I want to make sure @isuruf sees this before it goes in as well.

**Comment by ocefpaf** [comment]
So happy this is finally going in! I've been pushing for this since before Matt was a member :-)

**Comment by beckermr** [comment]
@ocefpaf did you want to enable this for all platforms? Right now this PR only does this for linux.

**Comment by isuruf** [comment]
In the last meeting, there was a request to add a `conda-forge.yml` option for this.

**Comment by jjhelmus** [comment]
Is there a method for feedstocks to opt-out of this?  There are a few packages that require tensorflow >=2 which is only available in `defaults`.  Turning this on without an option to opt-out makes this un-buildable until new tensorflow packages are added to conda-forge or the older packages are removed.

**Comment by beckermr** [comment]
ack - happy to do that 

the default needs to be set to yes iiuic, right

**Comment by beckermr** [comment]
thanks for the reminder all!

I will make a change to smithy. closing!

**Comment by jjhelmus** [comment]
> the default needs to be set to yes iiuic, right

That sounds great, default to strict but with an escape hatch for the handful of feedstock that need to opt-out

**Comment by isuruf** [comment]
You can read `conda-forge.yml` here.

**Comment by ocefpaf** [comment]
> @ocefpaf did you want to enable this for all platforms? Right now this PR only does this for linux.

I know. But it is a start ;-p
We need to get our act together on the scipy package to make the same on Windows. Not sure what are the hurdles on Macias.

**Comment by isuruf** [comment]
There are no hurdles on macOS.

**Comment by ocefpaf** [comment]
> There are no hurdles on macOS.

Screenshot_20200724-120007

File: recipe/run_conda_forge_build_setup_linux [hunk 1]
```diff
@@ -20,8 +20,12 @@ else
...
 fi
 
-# Need strict priority for pypy as defaults is not fixed
-if [[ "$CONFIG" == *_pypy* ]]; then
+# Need strict priority for
+# - pypy as defaults is not fixed
+# - cos7 as defaults is not fixed
+# but ppl can turn this off
+cfg_channel_priority=$(grep "channel_priority:" ${FEEDSTOCK_ROOT}/conda-forge.yml )
+if [[ ${cfg_channel_priority} == *channel_priority: strict* ]] || [[ -z ${cfg_channel_priority} ]]; then
```
Comment (isuruf) [inline recipe/run_conda_forge_build_setup_linux:28]: Can you use `shyaml` as in https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/master/recipe/run_conda_forge_build_setup_osx#L10

Comment (beckermr) [reply recipe/run_conda_forge_build_setup_linux:28]: that's clever! will change!

Comment (beckermr) [reply recipe/run_conda_forge_build_setup_linux]: done

**Comment by xhochy** [comment]
Please also make an announcement for this!

**Comment by beckermr** [comment]
ok @ocefpaf @isuruf @xhochy @jjhelmus @conda-forge/core any other comments here?

**Comment by beckermr** [comment]
doc pull here: https://github.com/conda-forge/conda-forge.github.io/pull/1112

**Comment by isuruf** [comment]
Can you send a PR for docs too if you have the time?

**Comment by beckermr** [comment]
LOL yup!

**Comment by isuruf** [comment]
This is failing on osx,

```
2020-07-24T15:54:51.6438850Z +++ shyaml get-value channel_priority strict
2020-07-24T15:54:51.6456570Z cat: /conda-forge.yml: No such file or directory
2020-07-24T15:54:51.6967720Z Error: invalid path 'channel_priority', can't query subvalue 'channel_priority' of a leaf (leaf value is None).
```

**Pull request merged**